contracts/examples/Venus.sol

```diff
@@ -18,69 +18,144 @@
+        (bytes32 srcChainId, bytes memory receiptRLP) = prover.validateReceipt(proof);
+        emit SuccessfulReceipt(srcChainId, receiptRLP);
     }
 
     /**
-     * * @notice Validates an event within a receipt proof using the ICrossL2Prover contract
-     * * @param receiptIndex The index of the receipt
-     * * @param receiptRLPEncodedBytes The RLP encoded receipt
-     * * @param proof The proof to validate
+     * * @notice Validates a generic event within a receipt proof using the ICrossL2Prover contract
+     * @notice reverts if the
+     * * @param proof The proof to validate, from which the rlp encoded bytes are fetched
      */
     function receiveEvent(
-        bytes calldata receiptIndex,
-        bytes calldata receiptRLPEncodedBytes,
         uint256 logIndex,
-        bytes calldata logBytes,
-        bytes calldata proof
+        bytes calldata proof,
+        address expectedEmitter,
+        bytes[] memory expectedTopics,
+        bytes calldata expectedUnindexedData
     ) external {
         // First we validate receipt to have a more helpful error message if the receipt itself is incorrect
 
-        if (!prover.validateReceipt(receiptIndex, receiptRLPEncodedBytes, proof)) {
+        // First, we validate the proof and get the rlp encoded bytes to prove the receipt is indeed in the MMPT of the
+        // counterparty chain's block
+        (bytes32 proofChainId, bytes memory receiptRLP) = prover.validateReceipt(proof);
+        if (proofChainId != chainId) {
+            revert invalidReceiptProof();
+        }
+
+        // Now that we have validated the receipt, we can trust the rlp encoded receipt bytes. Now we unpack the event
+        // data from these rlp encoded receipt bytes and validate it.
+        (address emittingContract, bytes[] memory topics, bytes memory unindexedData) =
+            prover.validateLog(logIndex, receiptRLP);
+
+        if (emittingContract != expectedEmitter) {
+            revert invalidEventSender();
+        }
+
+        if (!Bytes.equal(abi.encode(topics), abi.encode(expectedTopics))) {
+            revert invalidCounterpartyEvent();
+        }
+        if (!Bytes.equal(unindexedData, expectedUnindexedData)) {
+            revert invalidCounterpartyEvent();
+        }
+
+        emit ValidCounterpartyEvent(emittingContract, topics, expectedUnindexedData);
+    }
+
+    /**
+     * * @notice  Receive a specific houston transmission method. This is a useful example to illustrate how to
+     * deconstruct the event type and arguments within the evm layer
+     * @param logIndex The index within the tx of where the hello earth is logged in the transaction. Note: this is
+     * different than the index of the log within the block
+     * @param proof The proof to validate - returned by polymer proof api. This contains the proof to fetch a rlp
+     * encoded byte at a given index.
+     */
+    function receiveTransmissionEvent(uint256 logIndex, bytes calldata proof) external {
```

Comment:
Is this function now the same as receiveEvent, but with hardcoded expected values? Is it needed since it seems receiveEvent can replicate this functions behavior exactly?

Reply:
I guess it adds a tiny bit of extra "dapp" logic? Does it provide some other use?